### PR TITLE
Pop password from os_user params

### PR DESCRIPTION
**Pull request by emonty**
The password param conflicts with os-client-config's password grabbing.
The rest of the params really probably should also be popped - but
keeping this just to password for now. Will follow up with a change that
does an audit of all the os_ modules

Issue #4520

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks again to @emonty. This module is going into community review.

We notice that you are one of the maintainers (@emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/openstack/os_user.py
Comment (florianlechner) [inline cloud/openstack/os_user.py:158]: this line will fail with:
password = module.pop('password')
AttributeError: 'AnsibleModule' object has no attribute 'pop'

it should be:
password = module.params.pop('password')

**Review by florianlechner — CHANGES_REQUESTED** [review]

**Review by omgjlk — CHANGES_REQUESTED** [review]
This requires a code change.

**Comment by florianlechner** [comment]
yes it does, but merging the commit will cause a run time error.

**Comment by omgjlk** [comment]
That's my point, sorry I wasn't clear. I was using the github feature to provide a negative review, since this should not merge as is.

File: cloud/openstack/os_user.py
Comment (stephenpearson) [reply cloud/openstack/os_user.py:158]: Yes that worked for me (Ansible == 2.1.2.0 and passing auth via environment).  Thanks.

**Comment by gregdek** [comment]
Thanks @emonty for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

For help on how to do this cleanly please see http://docs.ansible.com/ansible/community.html#contributing-code-features-or-bugfixes

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@emonty A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
@emonty Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by mario-pranjic** [comment]
Hi,

I stumbled upon this issue today. 
os_project module can read env vars, but os-user fails.

Setting OpenStack (ocata) automation on latest CentOS. Using Ansible from CentoOS repo (2.2.1.0)

I see there are conflicts within this branch pending. Any chance this moves from full-stop soon?

**Pull request closed without merging**
